### PR TITLE
docs: improve Windows dev setup documentation

### PR DESCRIPTION
**Pull request by nigriMSFT**
Commit Message: docs: improve Windows dev setup documentation
Additional Description:

- Add H3 for each platform in step 1 to improve readability
- Improved Windows dev env in step 1
  - Formatting of install dependencies
  - Added cmd line examples for the additional config steps
  - Shuffled around install dependencies to decrease chances of PATH ordering issues (VSBT before MSYS2)

Risk Level: Low
Testing: N/A
Docs Changes: See description
Release Notes: N/A

### CODE REVIEW

**Comment by nigriMSFT** [comment]
@davinci26 @sunjayBhatia @wrowe this was helpful for me as I brought up a dev env (again). Let me know what ya think

**Comment by davinci26** [comment]
Looks awesome to me!

File: bazel/README.md [hunk 1]
```diff
@@ -97,34 +106,49 @@ for how to update or override dependencies.
...
-    of the installed MSYS2 `bash.exe` executable. Setting the `MSYS2_ARG_CONV_EXCL` environment
-    variable to a value of `*` is often advisable to ensure argument parsing in the MSYS2 shell
-    behaves as expected.
-
-    `Git` is required. The version installable via MSYS2 is sufficient.
+    ### Windows
+    On Windows, you'll need to install several dependencies manually.
 
-    Install the Windows-native [python3](https://www.python.org/downloads/), the POSIX flavor
+    [python3](https://www.python.org/downloads/): Specifically, the Windows-native flavor. The POSIX flavor
```
Comment (davinci26) [inline bazel/README.md:112]: I run into an issue with the store installation of python (https://github.com/msys2/MSYS2-packages/issues/1943).

Should we mention that python should be installed from the website? The reason I am bringing this up is if you type python on cmd/powershell the MSFT store pops up.

File: bazel/README.md [hunk 2]
```diff
@@ -97,34 +106,49 @@ for how to update or override dependencies.
...
+    For building with MSVC (the `msvc-cl` config option), you must install at least the VC++ workload.
     You may also download Visual Studio 2019 and use the Build Tools packaged with that
     installation. Earlier versions of VC++ Build Tools/Visual Studio are not recommended at this
     time. If installed in a non-standard filesystem location, be sure to set the `BAZEL_VC`
     environment variable to the path of the VC++ package to allow Bazel to find your installation
     of VC++. Use caution to ensure the `link.exe` that resolves on your PATH is from VC++ Build Tools and
     not MSYS2.
+    ```
+    set BAZEL_VC=%USERPROFILE%\VSBT2019\VC
+    set PATH=%PATH%;%USERPROFILE%\VSBT2019\VC\Tools\MSVC\14.26.28801\bin\Hostx64\x64
```
Comment (davinci26) [inline bazel/README.md:133]: nit: the path command might be a bit brittle if the version changes.

**Comment by lizan** [comment]
This looks good, shall we add how to run the build container locally too? I guess that will be the simplest way?

**Comment by sunjayBhatia** [comment]
@lizan I have a PR for that I had put off until now https://github.com/envoyproxy/envoy/pull/11363

File: bazel/README.md (hunk 1 above)
Comment (nigriMSFT) [reply bazel/README.md:112]: I can add an explicit warning for the store app like we have for MSYS2 flavor

File: bazel/README.md (hunk 2 above)
Comment (nigriMSFT) [reply bazel/README.md:133]: All of these commands are brittle to varying degrees. My main motivation was to provide an example of how the setup can be done. The details for each reader can certainly change. Maybe a disclaimer at the top for saying "the provided Windows commands are examples, ensure correct paths"?

Comment (davinci26) [reply bazel/README.md:133]: I agree, I think it is really valuable and it helped me a lot when I was looking into the commands.

**Pull request merged**